include/FanFicArchive.py

```diff
@@ -1,31 +1,36 @@
 from bs4 import BeautifulSoup; from urllib import request; from re import match; import requests
-def ao3_rip(ao3id): # downloads whole story in one go
-    if isinstance(ao3id, int): ao3id = str(ao3id)
-    else: assert isinstance(ao3id, str)
-    r = request.urlopen('http://archiveofourown.org/works/'+ao3id+'?view_adult=true&view_full_work=true').read()
-    text = BeautifulSoup(r, "lxml")
+
+parser = "lxml"
+
+def fanfic_req(link): return BeautifulSoup(request.urlopen(link).read(), parser)
+def fanfic_check(id, name):
+    if isinstance(id, int): id = str(id)
+    else: assert isinstance(id, str), "Error for" + name + ": id not int or str"
+
+def ao3_rip(ao3_id): # downloads whole story in one go
+    fanfic_check(ao3_id, "ao3_rip")
+    text = fanfic_req('http://archiveofourown.org/works/'+ao3_id+'?view_adult=true&view_full_work=true')
     tag_bar = text.find("dl", class_="stats").find_all("dd") # everything in the top bar
     tag_list = {}
     for i in tag_bar:
-        x = [j.get_text() for j in i.find_all("a", class_="tag")]
-        if x == []: x = [i.get_text().lstrip('\n').lstrip(' ').rstrip(' ').rstrip('\n')]
-        tag_list[i.get("class")[0]] = x
+        tag = [j.get_text() for j in i.find_all("a", class_="tag")]
+        if tag == []: tag = [i.get_text().lstrip('\n').lstrip(' ').rstrip(' ').rstrip('\n')]
+        tag_list[i.get("class")[0]] = tag
     tag_list['author'] = [i.get_text() for i in text.find_all("a", rel="author")]
     summary = text.find_all("div", class_="summary module", role="complementary") # just below top bar
     chapter = {}
     for i in text.find_all("div", class_="chapter", id=True): # separating each chapters
         x = i.find("h3", class_="title").find("a").get("href").split("/")[-1]
         chapter[i.get('id')] = [x, i]
-    return ao3id, tag_list, summary, chapter
-def ff_rip(ffid): # no image
-    if isinstance(ffid, int): ffid = str(ffid)
-    else: assert isinstance(ffid, str)
-    r = request.urlopen("https://www.fanfiction.net/s/" + ffid +"/1").read() # first chapter
-    text = BeautifulSoup(r, "lxml")
+    return ao3_id, tag_list, summary, chapter
+
+def ff_rip(ff_id): # no image
```

Comment:
The meaning of "no image" in this comment is unclear.

Reply:
Meaning: FanFiction does not have images to accompany the text.